internal/backend/remote-state/s3/backend_complete_test.go

```diff
@@ -650,8 +650,7 @@ aws_secret_access_key = DefaultSharedCredentialsSecretKey
 		tc := tc
 
 		t.Run(name, func(t *testing.T) {
-			oldEnv := servicemocks.InitSessionTestEnv()
-			defer servicemocks.PopEnv(oldEnv)
+			servicemocks.InitSessionTestEnv(t)
```

Comment:
Would you be able to discuss why we're doing this change here please? It's not very clear to me.

Reply:
Hey James! Of course! The changes is caused by the update of `aws-sdk-go-base` (which was required to deliver required functionality): the signature of the `servicemocks.InitSessionTestEnv` changed:

**_New_**
```go
// https://github.com/hashicorp/aws-sdk-go-base/blob/9e51f7a89d2d40ced63fd44fd7b6b4d29cc661aa/servicemocks/setup.go#L20-L24
func InitSessionTestEnv() (oldEnv []string) {
	oldEnv = StashEnv()
	os.Setenv("AWS_CONFIG_FILE", "file_not_exists")
	os.Setenv("AWS_SHARED_CREDENTIALS_FILE", "file_not_exists")

	return oldEnv
}
```

**_Status quo_**
```go
// https://github.com/hashicorp/aws-sdk-go-base/blob/192704b09f8b783d3990dbb4b9a975d5f7cb6b83/servicemocks/setup.go#L14-L20
func InitSessionTestEnv(t *testing.T) {
	StashEnv(t)
	t.Setenv("AWS_CONFIG_FILE", "file_not_exists")
	t.Setenv("AWS_SHARED_CREDENTIALS_FILE", "file_not_exists")
}
```

Reply:
Awesome, thanks for that 👍